lib/addons/argocd/index.ts

```diff
@@ -1,26 +1,359 @@
+import { HelmChart, KubernetesManifest, ServiceAccount } from "@aws-cdk/aws-eks";
+import { ManagedPolicy } from "@aws-cdk/aws-iam";
+import { SecretsManager } from "aws-sdk";
+
 import { ClusterAddOn, ClusterInfo, ClusterPostDeploy } from "../../stacks/cluster-types";
 import { Team } from "../../teams";
+import * as yaml from 'yaml';
+import { btoa } from '../../utils/string-utils';
+import { Construct } from "@aws-cdk/core";
+
+import bcrypt = require('bcrypt');
+
+
+export interface ArgoApplicationRepository {
```

Comment:
might want to move this to its own file to decouple `repo` concept from `Argo`

Reply:
I think it is a good idea, although who knows how they could diverge, e.g. options for Argo vs Flux. At present wego defined their own structure for it, so a bit late for convergence. However, agree that it is a generic structure. Will address.